src/libxtp/bulkesp.cc

```diff
@@ -0,0 +1,768 @@
+//                    if(fabs((*i)->x - 14.905)<0.002 && fabs((*j)->x - 14.266)<0.002){
+//                    //if(fabs((*i)->x - 14.266)<0.002){
+//                        cout<< (*i)->type << "\t" << (*i)->x << " " << (*i)->y << " " << (*i)->z <<endl;
+//                        cout<< (*j)->type << "\t" << (*j)->x << " " << (*j)->y << " " << (*j)->z <<endl;
+//                        cout<<"\tacceptDist:"<<acceptDist<<"\tscale:"<<scale<< "\tdist:"<<sqrt(distSq)<<endl;
+//                        cout << "\tPBC corrected dist:" <<distSq <<endl<<flush;
+//                        //exit(0);
+//                    }
+                    if (distSq <= acceptDist * acceptDist * scale * scale) {
+                        Bond nb; //bond goes
+                        nb.a = (*i); //from a
+                        nb.b = (*j); //to b
+                        nb.ba = v; //and has a (PBC corrected) vector ba (from b to a))
+                        nb.a_indx = i - _atoms.begin();
+                        nb.b_indx = j - _atoms.begin();
+                        bonds.push_back(nb);
+                    }
+                }
+            }
+            CTP_LOG(ctp::logDEBUG, *_log) << " BreakIntoMolecules(): " << bonds.size() << " bonds found.\n" << flush;
+            //        for (std::list<Bond>::iterator b = bonds.begin(); b != bonds.end(); ++b){
+            //            CTP_LOG(ctp::logDEBUG, *_log) << b->a_indx <<" - "<<b->b_indx<<"\t"<<b->ba<<"\t"<<sqrt(b->ba*b->ba)<< flush;
+            //        }
+
+
+            //now add all the atoms that have no bonds as a separate molecule
+            Bulkesp::Molecule leftover;
+            for (vector<ctp::QMAtom*>::iterator i = _atoms.begin(); i != _atoms.end(); ++i) {
+                bool found = false; //does this atom have bonds?
+                for (std::list<Bond>::iterator b = bonds.begin(); b != bonds.end(); ++b) {
+                    if (b->a == *i || b->b == *i) {
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found) //atom has no bonds
+                    leftover.atoms.push_back(*i);
+                leftover.atomIndeces.push_back(i - _atoms.begin());
+            }
+            if (leftover.atoms.size() > 0) {
+                mols.push_back(leftover);
+                CTP_LOG(ctp::logDEBUG, *_log) << " BreakIntoMolecules(): put " << leftover.atoms.size()
+                        << " Unbonded atoms into molecule " << mols.size() - 1 << ";\t" << bonds.size() << " bonds left.\nThey were:" << flush;
+                for(std::vector<ctp::QMAtom*>::iterator it = leftover.atoms.begin(); it != leftover.atoms.end(); ++it) {
+                    /* std::cout << *it; ... */
+                    CTP_LOG(ctp::logDEBUG, *_log) << (*it)->type << "\t" << (*it)->x << "\t" << (*it)->y << "\t" << (*it)->z << endl << flush;
+                }
+                exit(-1);
```

Comment:
what does this do?

Reply:
Breaks system into molecules following a bond graph. If there are any unconnected atoms, they get dumped into one "molecule"  because they are not likely to be embedded in anything after all the molecules are removed. So there is lots of gridpoints near them outside of VDW cutoffs and ESP can assign charges to them with little issue.